hazelcast-jclouds/src/test/java/com/hazelcast/jclouds/JCloudsDiscoveryStrategyTest.java

```diff
@@ -0,0 +1,100 @@
+import java.util.Set;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class JCloudsDiscoveryStrategyTest extends HazelcastTestSupport {
+    private static final Set<NodeMetadata> nodes = new HashSet<NodeMetadata>();
+    private static final Set<Address> addressesOfRunningInstances = new HashSet<Address>();
+
+    private static final int STARTING_PORT = 5701;
+    private static final int NUMBER_OF_NODES = 10;
+    private static final int NUMBER_OF_RUNNING_NODES = 7;
+
+    @BeforeClass
+    public static void start()
+            throws UnknownHostException {
+        for (int i = 0; i < NUMBER_OF_NODES; i++) {
+            HashSet<String> privateAdressses = new HashSet<String>();
+            privateAdressses.add("127.0.0." + (i+1));
+            NodeMetadata.Status status = NodeMetadata.Status.PENDING;
+            if (i < NUMBER_OF_RUNNING_NODES) {
+                status = NodeMetadata.Status.RUNNING;
+                addressesOfRunningInstances.add(new Address(privateAdressses.iterator().next(),0));
+            }
+            nodes.add(new NodeMetadataImpl("", "", "dummyId"+i, null, null, new HashMap<String, String>(), new HashSet<String>(), null, null, null, null, status, "",
+                    STARTING_PORT + i, privateAdressses, privateAdressses, null, "dummyHostName" + i));
+        }
+    }
+
+    @Test
+    public void testShouldDiscoverOnlyRunningNodes() {
+        ComputeServiceBuilder mockComputeServiceBuilder = mock(ComputeServiceBuilder.class);
+        doReturn(nodes).when(mockComputeServiceBuilder).getFilteredNodes();
+
+        JCloudsDiscoveryStrategy jCloudsDiscoveryStrategy = new JCloudsDiscoveryStrategy(mockComputeServiceBuilder);
+
+        Iterable<DiscoveryNode> runningNodes = jCloudsDiscoveryStrategy.discoverNodes();
+
+        for (DiscoveryNode node : runningNodes) {
+            assertTrue(addressesOfRunningInstances.contains(node.getPrivateAddress()));
+        }
+
+    }
+
+    @Test
+    public void testBuildCalled() {
+        ComputeServiceBuilder mockComputeServiceBuilder = mock(ComputeServiceBuilder.class);
+        JCloudsDiscoveryStrategy jCloudsDiscoveryStrategy = new JCloudsDiscoveryStrategy(mockComputeServiceBuilder);
+        jCloudsDiscoveryStrategy.start();
+
+        verify(mockComputeServiceBuilder).build();
+    }
+
+    @Test
+    public void testDestroyCalled() {
+        ComputeServiceBuilder mockComputeServiceBuilder = mock(ComputeServiceBuilder.class);
+        JCloudsDiscoveryStrategy jCloudsDiscoveryStrategy = new JCloudsDiscoveryStrategy(mockComputeServiceBuilder);
+        jCloudsDiscoveryStrategy.destroy();
+
+        verify(mockComputeServiceBuilder).destroy();
+    }
+
+    @Test(expected = HazelcastException.class)
+    public void testInvalidAddressShouldThrow() {
```

Comment:
I would suggest using the following naming

whenInvalidAddress_thenHazelcastException